core/app/[locale]/(default)/(faceted)/category/[slug]/page.tsx

```diff
@@ -77,297 +49,219 @@ const createCategorySearchParamsCache = cache(async (props: Props) => {
+      href: product.path,
+      image: product.defaultImage
+        ? { src: product.defaultImage.url, alt: product.defaultImage.altText }
+        : undefined,
+      price: pricesTransformer(product.prices, format),
+      subtitle: product.brand?.name ?? undefined,
+    }));
+  });
 
-async function getResetFiltersLabel(): Promise<string> {
-  const t = await getTranslations('Faceted.FacetedSearch');
+  const streamableTotalCount = Streamable.from(async () => {
+    const search = await streamableFacetedSearch;
 
-  return t('resetFilters');
-}
+    return search.products.collectionInfo?.totalItems ?? 0;
+  });
 
-async function getEmptyStateTitle(): Promise<string> {
-  const t = await getTranslations('Faceted.Category.Empty');
+  const streamablePagination = Streamable.from(async () => {
+    const search = await streamableFacetedSearch;
 
-  return t('title');
-}
+    return pageInfoTransformer(search.products.pageInfo);
+  });
 
-async function getEmptyStateSubtitle(): Promise<string> {
-  const t = await getTranslations('Faceted.Category.Empty');
+  const streamableFilters = Streamable.from(async () => {
+    const searchParams = await props.searchParams;
+    const searchParamsCache = await createCategorySearchParamsCache(categoryId);
+    const parsedSearchParams = searchParamsCache?.parse(searchParams) ?? {};
+    const search = await streamableFacetedSearch;
+
+    const allFacets = search.facets.items.filter(
+      (facet) => facet.__typename !== 'CategorySearchFilter',
+    );
+    const refinedFacets = search.facets.items.filter(
+      (facet) => facet.__typename !== 'CategorySearchFilter',
+    );
+
+    const transformedFacets = await facetsTransformer({
+      refinedFacets,
+      allFacets,
+      searchParams: { ...searchParams, ...parsedSearchParams },
+    });
+
+    const filters = transformedFacets.filter((facet) => facet != null);
+
+    const tree = categoryTree[0];
+    const subCategoriesFilters =
+      tree == null || tree.children.length === 0
+        ? []
+        : [
+            {
+              type: 'link-group' as const,
+              label: t('Category.subCategories'),
+              links: tree.children.map((child) => ({
+                label: child.name,
+                href: child.path,
+              })),
+            },
+          ];
+
+    return [...subCategoriesFilters, ...filters];
+  });
 
-  return t('subtitle');
-}
+  const streamableCompareProducts = Streamable.from(async () => {
+    const searchParams = await props.searchParams;
+    const customerAccessToken = await getSessionCustomerAccessToken();
+    const currencyCode = await getPreferredCurrencyCode();
 
-type SearchParams = Record<string, string | string[] | undefined>;
+    if (!productComparisonsEnabled) {
+      return [];
+    }
 
-interface Props {
-  params: Promise<{
-    slug: string;
-    locale: string;
-  }>;
-  searchParams: Promise<SearchParams>;
-}
+    const compareLoader = createCompareLoader();
```

Comment:
Should we create this outside the component? How come we have a function here vs exporting the loader?